src/include/daos_pool.h

```diff
@@ -449,6 +463,147 @@ int
 daos_pool_list_cont(daos_handle_t poh, daos_size_t *ncont,
 		    struct daos_pool_cont_info *cbuf, daos_event_t *ev);
 
+/** Logical function to apply to a container metadata key (current value <func> specified-value). */
+enum daos_pool_cont_filter_func {
+	/** key's current value equals filter-specified value parameter */
+	PCF_FUNC_EQ = 0,
+	/** key's current value does not equal filter-specified value parameter */
+	PCF_FUNC_NE,
+	/** key's current value is less than the filter-specified value parameter */
+	PCF_FUNC_LT,
+	/** key's current value is less than or equal to the filter-specified value parameter */
+	PCF_FUNC_LE,
+	/** key's current value is greater than the filter-specified value parameter */
+	PCF_FUNC_GT,
+	/** key's current value is greater than or equal to the filter-specified value parameter */
+	PCF_FUNC_GE,
+	/* future: add more functions */
+	PCF_FUNC_MAX
+};
+
+/** Convert logical function numeric identifier into printable string */
+static inline const char *
+daos_pool_cont_filter_func_str(enum daos_pool_cont_filter_func f)
+{
+	switch(f) {
+	case PCF_FUNC_EQ:
+		return "==";
+	case PCF_FUNC_NE:
+		return "!=";
+	case PCF_FUNC_LT:
+		return "<";
+	case PCF_FUNC_LE:
+		return "<=";
+	case PCF_FUNC_GT:
+		return ">";
+	case PCF_FUNC_GE:
+		return ">=";
+	default:
+		return "UNKNOWN";
+	}
+}
+
+/** container metadata key numeric identifier */
+enum daos_pool_cont_filter_key {
+	/** container metadata open time (UNIX time or HLC?). */
+	PCF_KEY_MD_OTIME = 0,
```

Comment:
Would it be worthwhile to document the type of the value accepted, that is, uint64_t or d_string_t, so that it would be clear which of pcfp_val64 and pcfp_valstr shall be used?

Reply:
will do